lib/Parse/ParseType.cpp

```diff
@@ -662,9 +663,7 @@ ParserStatus Parser::parseGenericArguments(SmallVectorImpl<TypeRepr *> &Args,
 
       Args.push_back(Ty.get());
       // Parse the comma, if the list continues.
-      if (!consumeIf(tok::comma))
-        break;
-    }
+    } while (consumeIf(tok::comma));
```

Comment:
This changes the behavior. `ElementContext` is a RAII object and it needs `consumeIf(tok::comma)` within the lifetime of it. Apparently, we don't have any test cases for it. Could you add something like:
```
typealias L = A<B, C>
```
to [test/Syntax/round_trip_parse_gen.swift](https://github.com/apple/swift/blob/master/test/Syntax/round_trip_parse_gen.swift#L146)?

Reply:
Sure!